eng/common/scripts/TypeSpec-Project-Process.ps1

```diff
@@ -57,59 +51,85 @@ function CreateUpdate-TspLocation() {
   $tspLocationYamlPath = Join-Path $packageDir "tsp-location.yaml"
   $tspLocationYaml = @{}
   if (Test-Path -Path $tspLocationYamlPath) {
-    $tspLocationYaml = Get-Content -Path $tspLocationYaml -Raw | ConvertFrom-Yaml
+    $tspLocationYaml = Get-Content -Path $tspLocationYamlPath -Raw | ConvertFrom-Yaml
+  }
+  else {
+    Write-Host "creating tsp-location.yaml in $packageDir"
   }
 
   # Update tsp-location.yaml
   $tspLocationYaml["commit"] = $CommitHash
-  $repo = ($RepoUrl -split "/")[3..4] -join "/"
   $tspLocationYaml["repo"] = $repo
   $tspLocationYaml["directory"] = $TypeSpecProjectDirectory
   $tspLocationYaml["additionalDirectories"] = $additionalDirs
   $tspLocationYaml |ConvertTo-Yaml | Out-File $tspLocationYamlPath
+  Write-Host "updated tsp-location.yaml in $packageDir"
+  return $packageDir
 }
 
-function Get-PackageDir() {
+function Get-PackageDir([System.Object]$tspConfig) {
+  $emitterName = ""
   if (Test-Path "Function:$GetEmitterNameFn") {
     $emitterName = &$GetEmitterNameFn
   }
+  else {
+    Write-Error "Missing $GetEmitterNameFn function in {$Language} SDK repo script."
+    exit 1
+  }
   $packageDir = ""
-  if ($tspConfigYaml["$emitterName"] -and $tspConfigYaml["$emitterName"]["package-dir"]) {
-    $packageDir = $tspConfigYaml["$emitterName"]["package-dir"];
+  if ($tspConfig["options"] -and $tspConfig["options"]["$emitterName"] -and $tspConfig["options"]["$emitterName"]["package-dir"]) {
+    $packageDir = $tspConfig["options"]["$emitterName"]["package-dir"]
   }
   else {
-    Write-Host "Missing package-dir in $emitterName options of tspconfig.yaml."
-    exit
+    Write-Error "Missing package-dir in $emitterName options of tspconfig.yaml."
+    exit 1
   }
   return $packageDir
 }
 
-$tspConfigContent = ""
+$repo = ""
+if ($RepoUrl) {
+  if ($RepoUrl -match "^https://github.com/(?<repo>[^/]*/azure-rest-api-specs(-pr)?).*") {
+    $repo = $Matches["repo"]
+  }
+  else {
+    Write-Host "Parameter 'RepoUrl' has incorrect value: $RepoUrl. It should be similar like 'https://github.com/Azure/azure-rest-api-specs'"
+    exit 1
+  }
+}
+
+$repoRootPath =  (Join-Path $PSScriptRoot .. .. ..)
+$repoRootPath = Resolve-Path $repoRootPath
+$repoRootPath = $repoRootPath -replace "\\", "/"
+$tspConfigPath = Join-Path $repoRootPath 'tspconfig.yaml'
 # example url of tspconfig.yaml: https://github.com/Azure/azure-rest-api-specs-pr/blob/724ccc4d7ef7655c0b4d5c5ac4a5513f19bbef35/specification/containerservice/Fleet.Management/tspconfig.yaml
-if ($TypeSpecProjectDirectory -match '^https://github.com/.*/tspconfig.yaml$') {
+if ($TypeSpecProjectDirectory -match '^https://github.com/(?<repo>Azure/azure-rest-api-specs(-pr)?)/blob/(?<commit>[0-9a-f]{40})/(?<path>.*)/tspconfig.yaml$') {
   try {
-    Invoke-WebRequest $TypeSpecProjectDirectory -OutFile $tspConfigContent -MaximumRetryCount 3
+    $TypeSpecProjectDirectory = $TypeSpecProjectDirectory -replace "github.com", "raw.githubusercontent.com"
```

Comment:
See https://github.com/Azure/azure-sdk/blob/main/eng/scripts/Generate-ReleaseNotes.ps1#L24 where I do a similar replacement in one line.